apm-agent-plugins/apm-scala-concurrent-plugin/src/main/java/co/elastic/apm/agent/scalaconcurrent/FutureInstrumentation.java

```diff
@@ -38,8 +39,8 @@
 public abstract class FutureInstrumentation extends TracerAwareInstrumentation {
 
     @SuppressWarnings("WeakerAccess")
-    public static final WeakConcurrentMap<Object, AbstractSpan<?>> promisesToContext =
-        new WeakConcurrentMap.WithInlinedExpunction<>();
+    public static final WeakMap<Object, AbstractSpan<?>> promisesToContext =
+        WeakConcurrent.buildMap();
```

Comment:
[minor] should probably be replaced by `WeakConcurrentSupplierImpl#createWeakSpanMap`.

Reply:
Let's do that in #2092